src/main/java/com/github/fishio/audio/Sound.java

```diff
@@ -0,0 +1,158 @@
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
+import com.github.fishio.logging.Log;
+import com.github.fishio.logging.LogLevel;
+
+/**
+ * Represents a playable sound.
+ * The bytes representing the sound are cached for quick playback.
+ */
+public class Sound {
+	/**
+	 * The size of the buffers for reading sounds.
+	 */
+	private static final int READ_BUFFER_SIZE = 16384;
+	
+	private AudioFormat format;
+	private boolean effect;
+	private byte[] contents;
+	
+	/**
+	 * Creates a new sound for the given InputStream.
+	 * 
+	 * @param inputStream
+	 * 		the stream to read the sound from.
+	 * @param mp3
+	 * 		if the stream is in mp3 format, this parameter should be true.
+	 * @param effect
+	 * 		<code>true</code> if this is a sound effect,
+	 * 		<code>false</code> if this is (background) music.
+	 * 
+	 * @throws IOException
+	 * 		If an IOException occurs while reading the given inputstream.
+	 * @throws UnsupportedAudioFileException
+	 * 		If the audio format is unsupported.
+	 */
+	public Sound(InputStream inputStream, boolean mp3, boolean effect)
+			throws IOException, UnsupportedAudioFileException {
+		this.effect = effect;
+		if (mp3) {
+			readMp3Stream(inputStream);
+		} else {
+			readAudioStream(inputStream);
+		}
+	}
+	
+	/**
+	 * Reads the given input stream as an audio file.
+	 * 
+	 * @param inputStream
+	 * 		the stream to read from.
+	 * 
+	 * @throws IOException
+	 * 		If an IOException occurs while reading the given input stream.
+	 * @throws UnsupportedAudioFileException
+	 * 		If the audio format is unsupported.
+	 */
+	protected void readAudioStream(InputStream inputStream) throws IOException, UnsupportedAudioFileException {
+		try (AudioInputStream ais = AudioSystem.getAudioInputStream(inputStream)) {
+			//Load the format
+			this.format = ais.getFormat();
+			
+			//Read the bytes from the audio file.
+			ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+	
+			int nRead;
+			byte[] data = new byte[READ_BUFFER_SIZE];
+	
+			while ((nRead = ais.read(data, 0, data.length)) != -1) {
+			  buffer.write(data, 0, nRead);
+			}
+	
+			buffer.flush();
+	
+			this.contents = buffer.toByteArray();
+		}
+	}
+	
+	/**
+	 * Read an input stream in mp3 format.
+	 * 
+	 * @param inputStream
+	 * 		the stream to read from.
+	 * 
+	 * @throws IOException
+	 * 		If an IOException occurs while reading the given input stream.
+	 * @throws UnsupportedAudioFileException
+	 * 		If the audio format is unsupported.
+	 */
+	protected void readMp3Stream(InputStream inputStream) throws IOException, UnsupportedAudioFileException {
+		try (AudioInputStream ais = AudioSystem.getAudioInputStream(inputStream)) {
+			//We want the file in the following format, so we can play it.
+			this.format = new AudioFormat(AudioFormat.Encoding.PCM_SIGNED, 
+					ais.getFormat().getSampleRate(),
+                    16,
```

Comment:
@MrCrime It's just github, but I will see if there is a bunch of spaces in there or something.

<layout